common/app/routes/challenges/redux/reducer.js

```diff
@@ -74,6 +75,7 @@ const mainReducer = handleActions(
       // used mainly to find code storage
       legacyKey: challenge.name,
       challenge: challenge.dashedName,
+      blockType: challenge.block,
```

Comment:
If I bring it in from the challengeSelector, I will get an argument name clash. I need both `block`'s to grab the `challengeSpec.name`

```js
const mapStateToProps = createSelector(
  challengeSelector,
  state => state.challengesApp.challenge,
  state => state.challengesApp.superBlocks,
  state => state.challengesApp.blockType,
  state => state.entities.block,
  (
    { challenge: { title = 'Learn to Code' } = {}, viewType, block },
    challenge,
    superBlocks = [],
    blockType,
    block
  ) => ({
    title,
    viewType,
    blockType,
    block,
    areChallengesLoaded: superBlocks.length > 0
  })
);
```